jax/_src/lax/control_flow/conditionals.py

```diff
@@ -732,9 +709,9 @@ def transposed(*args):
 
   return _make_closed_jaxpr(transposed, res_avals + jaxpr.out_avals)
 
-def _cond_transpose(cts, *args, branches, linear):
-  del linear  # could use for error checking, but see #14026
+def _cond_transpose(cts, *args, branches):
   index, *ops = args
+  assert type(index) is not ad.UndefinedPrimal
```

Comment:
This is an unrelated drive-by fix. This will never trigger during autodiff, but it can occur when using `linear_transpose` incorrectly (`cond` is not linear in the predicate!). This will at least help identify where the issue is coming from if anyone ever tries that again...